<Type Name="DirectoryNotFoundException" FullName="System.IO.DirectoryNotFoundException">
  <Metadata>
    <Meta Name="ms.openlocfilehash" Value="6ee62ee59bba76a52f5c9fc93dc7d11d9b060235" />
    <Meta Name="ms.sourcegitcommit" Value="d31dc2ede16f6f7bc64e90d9f897ff54c4e3869b" />
    <Meta Name="ms.translationtype" Value="HT" />
    <Meta Name="ms.contentlocale" Value="de-DE" />
    <Meta Name="ms.lasthandoff" Value="04/03/2018" />
  </Metadata>
  <TypeSignature Language="C#" Value="public class DirectoryNotFoundException : System.IO.IOException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit DirectoryNotFoundException extends System.IO.IOException" />
  <TypeSignature Language="DocId" Value="T:System.IO.DirectoryNotFoundException" />
  <TypeSignature Language="VB.NET" Value="Public Class DirectoryNotFoundException&#xA;Inherits IOException" />
  <TypeSignature Language="C++ CLI" Value="public ref class DirectoryNotFoundException : System::IO::IOException" />
  <AssemblyInfo>
    <AssemblyName>System.Runtime</AssemblyName>
    <AssemblyVersion>4.0.20.0</AssemblyVersion>
    <AssemblyVersion>4.1.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.1.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>mscorlib</AssemblyName>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.IO.IOException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute>
      <AttributeName>System.Runtime.InteropServices.ComVisible(true)</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary>
      <span data-ttu-id="72250-101">Die Ausnahme, die ausgelöst wird, wenn ein Teil einer Datei oder eines Verzeichnisses nicht gefunden werden kann.</span>
      <span class="sxs-lookup">
        <span data-stu-id="72250-101">The exception that is thrown when part of a file or directory cannot be found.</span>
      </span>
    </summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="72250-102">`DirectoryNotFoundException` verwendet die HRESULT COR_E_DIRECTORYNOTFOUND mit dem Wert 0 x 80070003.</span><span class="sxs-lookup"><span data-stu-id="72250-102">`DirectoryNotFoundException` uses the HRESULT COR_E_DIRECTORYNOTFOUND which has the value 0x80070003.</span></span> <span data-ttu-id="72250-103">Beachten Sie, dass `DirectoryNotFoundException` wird auch ausgelöst, wenn es sich bei COM-Interop-Programme das HRESULT STG_E_PATHNOTFOUND finden Sie unter dem der Wert 0 x 80030003 empfangen hat.</span><span class="sxs-lookup"><span data-stu-id="72250-103">Note that `DirectoryNotFoundException` is also thrown when COM interop programs see the HRESULT STG_E_PATHNOTFOUND, which has the value 0x80030003.</span></span>  
  
 <span data-ttu-id="72250-104">Wenn der Code nicht <xref:System.Security.Permissions.FileIOPermissionAttribute.PathDiscovery%2A> Berechtigung, die Fehlermeldung für diese Ausnahme nur Datei oder das Verzeichnis Namen anstelle von voll qualifizierten Pfade enthalten kann.</span><span class="sxs-lookup"><span data-stu-id="72250-104">If your code does not have <xref:System.Security.Permissions.FileIOPermissionAttribute.PathDiscovery%2A> permission, the error message for this exception may only contain file or directory names instead of fully qualified paths.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="72250-105">Im folgende Beispiel wird gezeigt, wie zu erzwingen und Wiederherstellen von einer <xref:System.IO.DirectoryNotFoundException>.</span><span class="sxs-lookup"><span data-stu-id="72250-105">The following example shows how to force and recover from a <xref:System.IO.DirectoryNotFoundException>.</span></span>  
  
 [!code-csharp[DirectoryNotFoundException#1](~/samples/snippets/csharp/VS_Snippets_CLR/DirectoryNotFoundException/cs/program.cs#1)]
 [!code-vb[DirectoryNotFoundException#1](~/samples/snippets/visualbasic/VS_Snippets_CLR/DirectoryNotFoundException/VB/DirectoryNotFoundexception.vb#1)]  
  
 ]]></format>
    </remarks>
    <altmember cref="T:System.Exception" />
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary>
          <span data-ttu-id="72250-106">Initialisiert eine neue Instanz der <see cref="T:System.IO.DirectoryNotFoundException" />-Klasse mit der zugehörigen Nachrichtenzeichenfolge.</span>
          <span class="sxs-lookup">
            <span data-stu-id="72250-106">Initializes a new instance of the <see cref="T:System.IO.DirectoryNotFoundException" /> class with its message string.</span>
          </span>
        </summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public DirectoryNotFoundException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.IO.DirectoryNotFoundException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; DirectoryNotFoundException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary>
          <span data-ttu-id="72250-107">Initialisiert eine neue Instanz der <see cref="T:System.IO.DirectoryNotFoundException" />-Klasse, bei der die Meldungszeichenfolge auf eine vom System gelieferte Meldung und HRESULT auf COR_E_DIRECTORYNOTFOUND festgelegt ist.</span>
          <span class="sxs-lookup">
            <span data-stu-id="72250-107">Initializes a new instance of the <see cref="T:System.IO.DirectoryNotFoundException" /> class with its message string set to a system-supplied message and its HRESULT set to COR_E_DIRECTORYNOTFOUND.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="72250-108">Dieser Konstruktor initialisiert die <xref:System.Exception.Message%2A?displayProperty=nameWithType> -Eigenschaft der neuen Instanz mit einer vom System gelieferten Meldung, die den Fehler beschreibt, wie z. B. "wurde nicht gefunden im angegebene Verzeichnis."</span><span class="sxs-lookup"><span data-stu-id="72250-108">This constructor initializes the <xref:System.Exception.Message%2A?displayProperty=nameWithType> property of the new instance to a system-supplied message that describes the error, such as "Could not find the specified directory."</span></span> <span data-ttu-id="72250-109">Diese Meldung berücksichtigt die aktuelle Systemkultur.</span><span class="sxs-lookup"><span data-stu-id="72250-109">This message takes into account the current system culture.</span></span>  
  
 <span data-ttu-id="72250-110">Die <xref:System.Exception.InnerException%2A?displayProperty=nameWithType> -Eigenschaft der neuen Instanz wird initialisiert, um `null`.</span><span class="sxs-lookup"><span data-stu-id="72250-110">The <xref:System.Exception.InnerException%2A?displayProperty=nameWithType> property of the new instance is initialized to `null`.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="72250-111">Im folgenden Codebeispiel wird veranschaulicht, wie zum Aufrufen der <xref:System.IO.DirectoryNotFoundException.%23ctor%2A> Konstruktor.</span><span class="sxs-lookup"><span data-stu-id="72250-111">The following code example demonstrates how to call the <xref:System.IO.DirectoryNotFoundException.%23ctor%2A> constructor.</span></span>  
  
 [!code-csharp[DirectoryNotFoundException#1](~/samples/snippets/csharp/VS_Snippets_CLR/DirectoryNotFoundException/cs/program.cs#1)]
 [!code-vb[DirectoryNotFoundException#1](~/samples/snippets/visualbasic/VS_Snippets_CLR/DirectoryNotFoundException/VB/DirectoryNotFoundexception.vb#1)]  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public DirectoryNotFoundException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.IO.DirectoryNotFoundException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; DirectoryNotFoundException(System::String ^ message);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="72250-112">Ein <see cref="T:System.String" />, der den Fehler beschreibt.</span>
          <span class="sxs-lookup">
            <span data-stu-id="72250-112">A <see cref="T:System.String" /> that describes the error.</span>
          </span>
          <span data-ttu-id="72250-113">Der Inhalt von <c>message</c> soll in verständlicher Sprache gehalten sein.</span>
          <span class="sxs-lookup">
            <span data-stu-id="72250-113">The content of <c>message</c> is intended to be understood by humans.</span>
          </span>
          <span data-ttu-id="72250-114">Der Aufrufer dieses Konstruktors muss sicherstellen, dass diese Zeichenfolge für die aktuelle Systemkultur lokalisiert wurde.</span>
          <span class="sxs-lookup">
            <span data-stu-id="72250-114">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="72250-115">Initialisiert eine neue Instanz der <see cref="T:System.IO.DirectoryNotFoundException" />-Klasse, bei der die Nachrichtenzeichenfolge auf <paramref name="message" /> und HRESULT auf COR_E_DIRECTORYNOTFOUND festgelegt wurde.</span>
          <span class="sxs-lookup">
            <span data-stu-id="72250-115">Initializes a new instance of the <see cref="T:System.IO.DirectoryNotFoundException" /> class with its message string set to <paramref name="message" /> and its HRESULT set to COR_E_DIRECTORYNOTFOUND.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="72250-116">Dieser Konstruktor initialisiert die <xref:System.Exception.Message%2A?displayProperty=nameWithType> -Eigenschaft der neuen Instanz unter Verwendung `message`.</span><span class="sxs-lookup"><span data-stu-id="72250-116">This constructor initializes the <xref:System.Exception.Message%2A?displayProperty=nameWithType> property of the new instance using `message`.</span></span>  
  
 <span data-ttu-id="72250-117">Die <xref:System.Exception.InnerException%2A?displayProperty=nameWithType> -Eigenschaft der neuen Instanz wird initialisiert, um `null`.</span><span class="sxs-lookup"><span data-stu-id="72250-117">The <xref:System.Exception.InnerException%2A?displayProperty=nameWithType> property of the new instance is initialized to `null`.</span></span>  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected DirectoryNotFoundException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.IO.DirectoryNotFoundException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; DirectoryNotFoundException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info">
          <span data-ttu-id="72250-118">Die <see cref="T:System.Runtime.Serialization.SerializationInfo" />, die die serialisierten Objektdaten für die ausgelöste Ausnahme enthält.</span>
          <span class="sxs-lookup">
            <span data-stu-id="72250-118">The <see cref="T:System.Runtime.Serialization.SerializationInfo" /> that holds the serialized object data about the exception being thrown.</span>
          </span>
        </param>
        <param name="context">
          <span data-ttu-id="72250-119">Der <see cref="T:System.Runtime.Serialization.StreamingContext" />, der die Kontextinformationen über die Quelle oder das Ziel enthält.</span>
          <span class="sxs-lookup">
            <span data-stu-id="72250-119">The <see cref="T:System.Runtime.Serialization.StreamingContext" /> that contains contextual information about the source or destination.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="72250-120">Initialisiert eine neue Instanz der <see cref="T:System.IO.DirectoryNotFoundException" />-Klasse mit der angegebenen Serialisierung und den angegebenen Kontextinformationen.</span>
          <span class="sxs-lookup">
            <span data-stu-id="72250-120">Initializes a new instance of the <see cref="T:System.IO.DirectoryNotFoundException" /> class with the specified serialization and context information.</span>
          </span>
        </summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public DirectoryNotFoundException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.IO.DirectoryNotFoundException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; DirectoryNotFoundException(System::String ^ message, Exception ^ innerException);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="72250-121">Die Fehlermeldung, in der die Ursache der Ausnahme erklärt wird.</span>
          <span class="sxs-lookup">
            <span data-stu-id="72250-121">The error message that explains the reason for the exception.</span>
          </span>
        </param>
        <param name="innerException">
          <span data-ttu-id="72250-122">Die Ausnahme, die die Ursache der aktuellen Ausnahme ist.</span>
          <span class="sxs-lookup">
            <span data-stu-id="72250-122">The exception that is the cause of the current exception.</span>
          </span>
          <span data-ttu-id="72250-123">Wenn der <c>innerException</c>-Parameter nicht <see langword="null" /> ist, wird die aktuelle Ausnahme in einem <see langword="catch" />-Block ausgelöst, der die innere Ausnahme behandelt.</span>
          <span class="sxs-lookup">
            <span data-stu-id="72250-123">If the <c>innerException</c> parameter is not <see langword="null" />, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="72250-124">Initialisiert eine neue Instanz der <see cref="T:System.IO.DirectoryNotFoundException" />-Klasse mit einer angegebenen Fehlermeldung und einem Verweis auf die innere Ausnahme, die diese Ausnahme ausgelöst hat.</span>
          <span class="sxs-lookup">
            <span data-stu-id="72250-124">Initializes a new instance of the <see cref="T:System.IO.DirectoryNotFoundException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="72250-125">Eine Ausnahme, die als direktes Ergebnis einer vorhergehenden Ausnahme ausgelöst wird, muss in der <xref:System.Exception.InnerException%2A>-Eigenschaft über einen Verweis auf die vorhergehende Ausnahme verfügen.</span><span class="sxs-lookup"><span data-stu-id="72250-125">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="72250-126">Die <xref:System.Exception.InnerException%2A>-Eigenschaft gibt den gleichen Wert zurück, der an den Konstruktor übergeben wird, oder `null`, wenn die <xref:System.Exception.InnerException%2A>-Eigenschaft den Wert der inneren Ausnahme nicht an den Konstruktor übergibt.</span><span class="sxs-lookup"><span data-stu-id="72250-126">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or `null` if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="72250-127">In der folgenden Tabelle werden die anfänglichen Eigenschaftenwerte für eine <xref:System.IO.DirectoryNotFoundException>-Instanz aufgeführt.</span><span class="sxs-lookup"><span data-stu-id="72250-127">The following table shows the initial property values for an instance of <xref:System.IO.DirectoryNotFoundException>.</span></span>  
  
|<span data-ttu-id="72250-128">Eigenschaft</span><span class="sxs-lookup"><span data-stu-id="72250-128">Property</span></span>|<span data-ttu-id="72250-129">Wert</span><span class="sxs-lookup"><span data-stu-id="72250-129">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="72250-130">Der Verweis auf die interne Ausnahme.</span><span class="sxs-lookup"><span data-stu-id="72250-130">The inner exception reference.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="72250-131">Die Zeichenfolge der Fehlermeldung.</span><span class="sxs-lookup"><span data-stu-id="72250-131">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Exception" />
      </Docs>
    </Member>
  </Members>
</Type>